include/boost/math/distributions/exponential.hpp

```diff
@@ -127,6 +127,24 @@ inline RealType pdf(const exponential_distribution<RealType, Policy>& dist, cons
    return result;
 } // pdf
 
+template <class RealType, class Policy>
+inline RealType logpdf(const exponential_distribution<RealType, Policy>& dist, const RealType& x)
+{
+   BOOST_MATH_STD_USING // for ADL of std functions
+
+   static const char* function = "boost::math::logpdf(const exponential_distribution<%1%>&, %1%)";
+
+   RealType lambda = dist.lambda();
+   RealType result = 0;
```

Comment:
We're going to have issues with copying over the special case handling in all of these I think, we may need to think about this, but to be consistent with taking the log of the pdf, we should start with result = -INF for these special cases.